docs/code/ConcolicFuzzer.py

```diff
@@ -2158,17 +2158,10 @@ def add_child(self, elt, i, cc, string):
 
         # first we update our smt decision
         if bit == self.tree.yes_bit:  # yes, which means the smt can be used as is
-            if self.smt is not None:
-                assert self.smt == child_node._smt_val
-            else:
-                self.smt = child_node._smt_val
+            self.smt = child_node._smt_val
```

Comment:
When would a child node have a different `_smt_val` than the current `self.smt`?

Reply:
The assertion was violated after I've implemented some function summaries, and it seems to work with the fix.

Reply:
To reproduce, using fuzzingbook's find summary: https://gist.github.com/leonbett/69b29cbe9d3602f8f59c431b5d5782a5

Reply:
What I meant is that `TraceNode.smt` value is only filled when a child was added right? otherwise `smt` is `None`. So, for this assumption to be violated, a previous child should have been added with a different `smt`. How would that happen as the children represent decisions taken on the same smt?

Reply:
I think the problem is that __getitem__ in ConcolicTracer currently concretizes its argument, leading to a mismatch. I'll try to tweak the exercise program and see if it works. Thanks.